### PR TITLE
#6: One page checkout

### PR DESCRIPTION
**Pull request by tanvu-epi**
Feature OnePageCheckout 
Fix BuyNow click

### CODE REVIEW

**Comment by lunchin** [comment]
This looks good.

Some initial feedback would be have an option to use the shipping address for the billing address along with saved address, and new address

Shipments don't seem to be added correctly when purchasing

Subscriptions header missing.  Also should add additional options for creating subscription, max number of instance and start date.

**Comment by tanvu-epi** [comment]
> 
> Shipments don't seem to be added correctly when purchasing
> 

Please describe more clearly. I see that the shipment in Commerce Manager is still correct

File: src/Foundation/Features/Checkout/Checkout.cshtml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+@using (Html.BeginForm("PlaceOrder", "Checkout", FormMethod.Post, new { @class = "row jsCheckoutForm", id = "jsCheckoutForm", novalidate = "novalidate" }))
+{
+    @Html.AntiForgeryToken()
+    <div class="col-md-8 col-xs-12">
+        <div class="card items">
+            <div class="card-body">
+                @Html.Partial("ShippingInformation", Model)
+                <br />
+                @Html.Partial("SingleAddress", Model)
```
Comment (lunchin) [inline src/Foundation/Features/Checkout/Checkout.cshtml]: I think the ShippingInformation and Single Address views need to be combined to properly support multiple shipments.  The new view should loop through each shipment like before.  Please update the view with the following

- Add a split shipment button after Quantity text box that is only enabled when quantity is greater than one
- When the split button is clicked a new shipment will be created with 1 quantity removed the original shipment and added to the new shipment.
- Each shipment should have an address selector for that particular shipment.  Please add border around each shipment and separate shipping section with a border as well to show to user the section is for shipping information
- Remove the use billing address as shipping address
- Add a button for for entering new address to update the order when the new address is entered
- When changing address in the drop down make sure to update the order with the new address
- With these changes shipping address information should be updated before going to the next billing section

File: src/Foundation/Features/Checkout/Checkout.cshtml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                <hr />
+                @Html.Partial("BillingInformation", Model)
```
Comment (lunchin) [inline src/Foundation/Features/Checkout/Checkout.cshtml]: Billing Information should be updated 

- Billing Information should be its own section with a border to distinguish Billing Information
- If there is only one shipment add a new radio button to Use Shipping Address for Billing Address as the first option
- Separate the Coupon partial view and move to the top of the checkout before the shipping information

File: src/Foundation/Features/Checkout/Checkout.cshtml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            </div>
+        </div>
+        <br />
+        <div class="card items">
+            <div class="card-body">
+                @Html.Partial("Subscription", Model)
```
Comment (lunchin) [inline src/Foundation/Features/Checkout/Checkout.cshtml]: Please add the following to subscriptions

- Change checkbox label to create recurring order.
- Remove cycle mode as we will always use days
- Update the label to and text box for cycle length to be Every (Text Box) days
- Add new datetime textbox for choosing the end date.  Can be nullable
- Update the start date to be x number of days from current order

File: src/Foundation/Features/Checkout/CheckoutMethod.cshtml [hunk 4]
```diff
@@ -1,7 +1,5 @@
 ﻿@model CheckoutMethodViewModel
 
-@Html.Partial("_CheckoutStatusBar", 1)
```
Comment (lunchin) [inline src/Foundation/Features/Checkout/CheckoutMethod.cshtml]: Delete the file if it is not used.

**Pull request merged**